File: JavaScript_Courses/JavaScript_Basics/JS_BP_2022_10/04_For_Loop_Exercise/08_Tennis_Ranklist.js
function tennisRanklist(input) {
    //     Има три варианта за завършване на турнир:
    // W - ако е победител получава 2000 точки  // winner
    // F - ако е финалист получава 1200 точки   // final
    // SF - ако е полуфиналист получава 720 точки  // semi-final
    let startingPoints = Number(input[1]);
    let totalPointsGained = startingPoints;
    let countsW = Number(input[0]);
    let countsW = 0;

    countsW++;

    // •	"Final points: {брой точки след изиграните турнири}"
    // •	"Average points: {средно колко точки печели за турнир}"
    // •	"{процент спечелени турнири}%"

    console.log(`Final points: ${totalPoints}`);
    console.log(`Average points: ${averagePoints}`);  // averagePoints = totalPointsGained / tournamentCounts;
    // totalPointsGained = totalPoints - startingPoints  // input[1]
    console.log(`${percentTourWins}%`);  // percentTourWins = countsW / countsTotal;
}


tennisRanklist([
    "5",
    "1400",
    "F",
    "SF",
    "W",
    "W",
    "SF",
]);

File: Python_Courses/Python_Basics/Python_PB_2024_01/04_For_Loop_Exercise/08_Tennis_Ranklist.py
tournaments_count = int(input())
starting_points = int(input())

gained_points = 0
count_wins = 0
for _ in range(tournaments_count):
    result = input()  # "W", "F", or "SF"
    if result == "W":
        gained_points += 2000
        count_wins += 1
    elif result == "F":
        gained_points += 1200
    elif result == "SF":
        gained_points += 720

total_points = starting_points + gained_points
average_points = gained_points / tournaments_count
percent_wins = count_wins / tournaments_count

print(f"Final points: {total_points}")
print(f"Average points: {int(average_points)}")
print(f"{percent_wins :.2%}")

